### PR TITLE
[SERVER-1169] Add details on how to configure an external vault instance

### PR DESCRIPTION
**Pull request by lloydChris**
# Description
Server now supports Externalizing Vault.  Add documentation so people know how to do it.

# Reasons
https://circleci.atlassian.net/browse/SERVER-1169

Please critique the formatting.  The internal/external part is particularly bad

### CODE REVIEW

**Review by zawzawaung-prog — APPROVED** [review]

**Comment by zawzawaung-prog** [comment]
Good

File: jekyll/_cci2/server-3-install.adoc [hunk 1]
```diff
@@ -143,6 +143,41 @@ The domain you selected for your CircleCI installation is the Homepage URL and *
 
 NOTE: If using GitHub Enterprise, you will also need a personal access token and the domain name of your GitHub Enterprise instance. You must also enable HTTP API Rate Limiting from the management console.
 
+### Vault
+*Internal* will deploy the default Vault instance inside the CircleCI K8s namespace.  The application will be automatically configured.
+*External* will not install the default Vault instance with the CircleCI application.  Select this option if using a Vault instance ou.  Will require the user to configure the following settings:
+
+. URL: ex. `http://vault:8200`
+. Transit Path: The name of the secrets engine.  The default is `transit`. See the https://www.vaultproject.io/docs/secrets/transit#setup[Vault documentation] for more details.
```
Comment (nsmith5) [inline jekyll/_cci2/server-3-install.adoc:151]: Nit: "name" feels a little ambiguous in that it doesn't really match anything in the Hashicorp docs

```suggestion
. Transit Path: The path of the transit secrets engine.  The default is `transit`. See the https://www.vaultproject.io/docs/secrets/transit#setup[Vault documentation] for more details.
```

**Review by nsmith5 — APPROVED** [review]
Looks good to me. One nit about engine wording

File: jekyll/_cci2/server-3-install.adoc [hunk 2]
```diff
@@ -143,6 +143,41 @@ The domain you selected for your CircleCI installation is the Homepage URL and *
 
 NOTE: If using GitHub Enterprise, you will also need a personal access token and the domain name of your GitHub Enterprise instance. You must also enable HTTP API Rate Limiting from the management console.
 
+### Vault
+*Internal* will deploy the default Vault instance inside the CircleCI K8s namespace.  The application will be automatically configured.
+*External* will not install the default Vault instance with the CircleCI application.  Select this option if using a Vault instance ou.  Will require the user to configure the following settings:
```
Comment (MikeyYeahYeah) [inline jekyll/_cci2/server-3-install.adoc:148]:
```
Select this option if using a Vault instance ou.
```
Is the "ou" a typo?

File: jekyll/_cci2/server-3-install.adoc (hunk 1 above)
Comment (lloydChris) [reply jekyll/_cci2/server-3-install.adoc]: Fixed, thanks!

**Comment by annabaker** [comment]
Just looking for clarification on where this should go/when via branch naming convention, as some content has been opened up against master instead of the 3.1 branch for newer PRs: https://github.com/circleci/circleci-docs#branch-nomenclature-for-subsequent-server-releases

**Review by MikeyYeahYeah — APPROVED** [review]

**Comment by lloydChris** [comment]
> Just looking for clarification on where this should go/when via branch naming convention, as some content has been opened up against master instead of the 3.1 branch for newer PRs: https://github.com/circleci/circleci-docs#branch-nomenclature-for-subsequent-server-releases

Good catch.  I updated it to point at the 3.1 branch

**Comment by lloydChris** [comment]
Replaced by: https://github.com/circleci/circleci-docs/pull/5428

**Pull request closed without merging**